flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/RestHandlerConfiguration.java

```diff
@@ -96,13 +105,20 @@ public static RestHandlerConfiguration fromConfiguration(Configuration configura
 
         final boolean webSubmitEnabled = configuration.getBoolean(WebOptions.SUBMIT_ENABLE);
         final boolean webCancelEnabled = configuration.getBoolean(WebOptions.CANCEL_ENABLE);
+        final boolean webRescaleSupported =
```

Comment:
Note: This may not work as we'd like it to. If the user has configured the adaptive scheduler, but submits a batch job, then the buttons will still be shown.

Reply:
Hmm, that's a good point; I'm wondering if there's anything we can do about it since this is a cluster-wide configuration. Maybe we can check the UI level for `webRescaleSupported && job.isStreaming()`? WDYT?

Reply:
Do we expose batch/streaming type in the REST API somewhere? I couldn't find that.

long-term we may want to add a per-job config handler, that could report whether checkpoints/rescaling/whatever job-specific thing is supported.

Reply:
>  If the user has configured the adaptive scheduler, but submits a batch job, then the buttons will still be shown.

let's take this as a limitation for now, and I'll create a follow up ticket to address this as the fix allows tweaking JobDetails rest API endpoint